pallets/liquid-staking/src/types.rs

```diff
@@ -101,19 +101,19 @@ impl<Balance: BalanceT + FixedPointOperand> MatchingLedger<Balance> {
         Ok(())
     }
 
-    fn sub_stake_amount(&mut self, amount: Balance) -> DispatchResult {
+    pub fn sub_stake_amount(&mut self, amount: Balance) -> DispatchResult {
         self.total_stake_amount.total = self
             .total_stake_amount
-            .total
+            .free()?
```

Comment:
in `consolidate_stake`, `total_stake_amount.total` and `total_stake_amount.reserve` will deduct a same amount, if change to 'free()', then `reserve` has been deduct here, it's better to use `total`.

Reply:
yes I think I did a bad change, I'd like to deduct the free part but this seems to be a wrong way